src/prefect/utilities/storage.py

```diff
@@ -79,7 +79,7 @@ def extract_flow_from_file(
             contents = f.read()
 
     elif file_contents is not None:
-        contents = file_contents
+        contents = file_contents.encode("utf-8")
```

Comment:
I'm not sure we want to do this. I think we want `exec` to just accept the raw bytes without choosing the encoding. Was there a type-check error or something?

Reply:
Yes, mypy found: 
```
src/prefect/utilities/storage.py:82: error: Incompatible types in assignment (expression has type "str", variable has type "bytes"
```

Reply:
Ah I see here we're then using `bytes` for `contents` consistently. All strings _should_ be utf-8 encoded already and you can just call `.encode()` here. I'd leave the encoding up to Python. Alternatively, we can update `file_contents` to `Optional[Union[str, bytes]]` instead.

Reply:
I went for the first option in [60c51bd](https://github.com/PrefectHQ/prefect/pull/6375/commits/60c51bd0852b764eec9f44c0b5b5f7d81b9f6be2).